### DISCUSSION TITLE
periodicity with a rotated columnar grain

### DISCUSSION THREAD

**Discussion by felix0685** [neperfepx/neper#331] [Q&A] [answered]
Hello,

I am working on a project regarding metal 3d printing. We have some samples printed under 0°, 45° and 90° orientation to the scanning direction and have some EBSD scans made. The result of the EBSD shows we have a columnar grain structure and a preferred crystal orientation. Now we can easily create a model for the 0° and 90° case.

however I am struggling with the 45° orientation a little bit right now. With this command I can generate a rotated grain

`neper -T -n 100 -dim 3 -domain "cube(1,1,1):rotate(0,1,0,$angle)" -morpho "columnar(x)" -o fileName -format geo`

and it results in this
![columnar_3](https://user-images.githubusercontent.com/94524623/142172542-95b20bf8-206a-499f-9f21-60ff48a745de.png)

 using: 

`neper -T -n 100 -dim 3 -domain "cube(1,1,1):rotate(0,1,0,45)" -morpho "columnar(x)" -periodicity y -o columnar_4`

results in a periodicity in Y-direction:

![columnar_4](https://user-images.githubusercontent.com/94524623/142173163-6dafafce-50db-482d-abcd-3ee0b10845f1.png)

So I do understand that the periodicity in X- and Z-direction doesn't work as as the cube is rotated so my Question would be how to achieve the periodicity along the other two faces?

Thank you, Felix

**Comment by smallrain1996** [comment]
Your rotation axis is the y-axis, which can be considered as before and after rotation, all y-coordinates are constant, so only the periodicity of the y-axis can be preserved. If you try to set periodicity on x,z, this seems logically incomprehensible and therefore an error is reported. Similarly, if your rotation axis is (1,1,1), no periodicity boundary condition can be added.

**Comment by rquey** [comment, marked as answer]
I think @smallrain1996 is right in that you cannot apply periodicity in a direction along which the domain itself is not periodic.

A possibility may be to rotate the seeds... For the non-periodic case, it will work. For the fully periodic case, I am not entirely sure (because of the columnar structure). Worth the try!

**Reply by felix0685** [reply]
Thank you for the reply. As I said I did understand I can't have periodicity with X, and Z due to the rotation but was hoping there would be a way to define a new axis to preserve periodicity.
For the non-periodic case I am fine with the rotated result, as the current workflow is to import it into Abaqus and in combination with DAMASK run the simulation, so I have the possibility to still rotate and translate the cube, but I need the periodicity to work out as well.

I will try to look into rotating the seeds.
